src/CMakeLists.txt

```diff
@@ -143,6 +143,17 @@ if(APPLE AND (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR CMAKE_CXX_COMPILER_
         list(REMOVE_DUPLICATES CMAKE_CXX_IMPLICIT_LINK_LIBRARIES)
 endif()
 
+# Check for base threading library
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+find_package(Threads REQUIRED)
+if(${CMAKE_THREAD_LIBS_INIT} MATCHES "-pthread")
```

Comment:
Ha! it seems I was confused by errors reported by travis, and that adding -pthread to LIBS is well managed by cmake (the issue was that darktable-rs-identify must be explicitely linked with thread library/flag as it does not use LIBS).

So this ugly test is now removed, and travis seems happy with the regular branch, and a test branch that use the cmake 3.1 option to use the flag to also test this case (https://travis-ci.org/mazhe/darktable/builds/186924244). My FreeBSD system is also happy with this code, the only configuration I cannot test is win64.